addons/sale_timesheet/controllers/pivot.py

```diff
@@ -0,0 +1,44 @@
+import json
+import math
+
+from odoo import http, _
+from odoo.http import request
+from odoo.tools import float_round
+from odoo.addons.web.controllers.pivot import TableExporter
+
+
+class PivotExporter(TableExporter):
+
+    @http.route('/web/pivot/export_xlsx', type='http', auth="user")
+    def export_xlsx(self, data, **kw):
+        jdata = self.convert_columns(data)
+        return super().export_xlsx(json.dumps(jdata), **kw)
+
+    def convert_columns(self, data):
+        """
+        Backend data are all in hours. Widget converts the uom (hours > day) only for the frontend pivot view
+        The function is to handle it error-proof for downloading xlsx feature
+        """
+
+        DAY_IN_HOURS = request.env['uom.uom'].with_context(lang='en_US').search([('name', '=', 'Hours')]).factor
+        dict_data = json.loads(data)
+        indexes = []
+        index = 0
+        if dict_data['model'] == 'timesheets.analysis.report':
+            for col in dict_data['measure_headers']:
+                if _('Days Spent').lower() in col['title'].lower():
```

Comment:
I think this new translation in sale_timesheet could be different than the translation of "Days Spent" in hr_timesheet. So this would add the constraint that all the translation in sale_timesheet should be for the fix to work the same as the translation in (I guess) hr_timesheet.